python_modules/libraries/dagster-components/dagster_components/core/schema/objects.py

```diff
@@ -11,26 +11,22 @@
 )
 from dagster._core.definitions.definitions_class import Definitions
 from dagster._record import replace
+from pydantic import BaseModel
 
-from dagster_components.core.schema.base import ComponentSchemaBaseModel
+from dagster_components.core.schema.base import ResolvableModel
 from dagster_components.core.schema.metadata import ResolvableFieldInfo
 from dagster_components.core.schema.resolver import TemplatedValueResolver
 
 
-def _post_process_key(resolved: Optional[str]) -> Optional[AssetKey]:
-    return AssetKey.from_user_string(resolved) if resolved else None
-
-
-class OpSpecBaseModel(ComponentSchemaBaseModel):
+class OpSpecModel(ResolvableModel["OpSpecModel"]):
     name: Optional[str] = None
     tags: Optional[dict[str, str]] = None
 
+    def resolve(self, resolver: TemplatedValueResolver) -> "OpSpecModel":
+        return OpSpecModel(**resolver.resolve_obj(self.model_dump(exclude_unset=True)))
 
-class AssetAttributesModel(ComponentSchemaBaseModel):
-    key: Annotated[
-        Optional[str],
-        ResolvableFieldInfo(output_type=AssetKey, post_process_fn=_post_process_key),
-    ] = None
+
+class _ResolvableAssetAttributesModel(BaseModel):
```

Comment:
might want to use the word `Mixin` here to make it clear that this not meant to be used independently